### PR TITLE
Added Pokedex Plugin

### PR DESCRIPTION
**Pull request by ValynEritai**
File is located at chat-bot/plugins/pokedex.py. Should be complete for now, Bugs should be minimal, if needed I will fix them.

Feel free to whip up a dev copy to test. :)

### CODE REVIEW

File: chat-bot/plugins/pokedex.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            {
+                'name': '!pokedex <pokedex_number>',
+                'description': 'Gives you information about a pokemon. Use National Dex number.'
+            }
+        ]
+        return commands
+
+    async def get_json(self, number):
+        data = json.loads('http://pokeapi.co/api/v2/pokemon-species/', number/)
+		return data
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: Bad indent.

File: chat-bot/plugins/pokedex.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import logging
+import asyncio
+import os
+import json
+import re
+import aiohttp
+import html
+
+log = logging.getLogger('discord')
+MAL_USERNAME = os.getenv('MAL_USERNAME')
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: Unused vars. You don't need that

File: chat-bot/plugins/pokedex.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+from plugin import Plugin
+import logging
+import asyncio
+import os
+import json
+import re
+import aiohttp
+import html
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: We don't need that package

File: chat-bot/plugins/pokedex.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+log = logging.getLogger('discord')
+MAL_USERNAME = os.getenv('MAL_USERNAME')
+MAL_PASSWORD = os.getenv('MAL_PASSWORD')
+
+class AnimuAndMango(Plugin):
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: Pokedex should be the name of that Class

File: chat-bot/plugins/pokedex.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        commands = [
+            {
+                'name': '!pokedex <pokedex_number>',
+                'description': 'Gives you information about a pokemon. Use National Dex number.'
+            }
+        ]
+        return commands
+
+    async def get_json(self, number):
+        data = json.loads('http://pokeapi.co/api/v2/pokemon-species/', number/)
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: json.loads don't get any file. Look at http://aiohttp.readthedocs.org/en/stable/client.html?highlight=get :wink:

File: chat-bot/plugins/pokedex.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            return
+
+        log.info('{}#{}@{} >> {}'.format(
+            message.author.name,
+            message.author.discriminator,
+            message.server.name,
+            message.clean_content
+        ))
+
+        number = check.groups()
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: This shouldn't be named number. And should be equal to check.groups[1]

File: chat-bot/plugins/pokedex.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        ]
+        return commands
+
+    async def get_json(self, number):
+        data = json.loads('http://pokeapi.co/api/v2/pokemon-species/', number/)
+		return data
+
+    async def on_message(self, message):
+
+        rule = r'!(pokedex) (.*)'
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: Maybe use a stronger regex like `r'^!pokedex (.*)$'`

**Comment by ValynEritai** [comment]
Ack, sorry. My mistake, I'll go fix now. Haven't touched python for a few months, so i forgot how finicky it is with indents. Also the vars I over looked and I realised the class name is unchanged.

File: chat-bot/plugins/pokedex.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+        data = await self.get_json(nature, number)
+        if data == '':
+            await self.mee6.send_message(
+                message.channel,
+                'Pokemon not found. Try again.'
+            )
+            return
+
+		msg = ('Pokemon Number', data['id'], ": ", data['names']['name'], '\n'}
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: Bad indent. You are using a tab indent. Try to configure your text editor to use a 4 space indent.

File: chat-bot/plugins/pokedex.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+		if data['gender_rate'] != -1: {
+		msg += ('Genderless'\n')}
+		else: {
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: why `{` :p ?

File: chat-bot/plugins/pokedex.py [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+		msg += ('Gender Ratio:', data['gender_rate'] * 12.5, '% Female, ', 100 - (data['gender_rate'] * 12.5), '% Male.\n'})
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: you don't need/have to use `(...)`

**Comment by cookkkie** [comment]
GL, HF mate! :smiley:

**Comment by ValynEritai** [comment]
Okay, have a look see now. If there's more wrong with it, get back to me and I'll fix again. Got plenty of time. I'm just rusty. haha

File: chat-bot/plugins/pokedex.py [hunk 11]
```diff
@@ -5,32 +5,31 @@
...
-class AnimuAndMango(Plugin):
+
+class Pokedex(Plugin):
 
     fancy_name = 'Pokedex'
 
     async def get_commands(self, server):
         commands = [
-            {
+		    {
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: Mixed indent here. Do not use tabs. Use 4 spaces instead.

**Comment by ValynEritai** [comment]
Dammit, I just realised there's a bad string. I'm gonna put up a fix really quick.

File: chat-bot/plugins/pokedex.py [hunk 12]
```diff
@@ -54,29 +53,29 @@ class AnimuAndMango(Plugin):
...
-		else: {
-		msg += ('Gender Ratio:', data['gender_rate'] * 12.5, '% Female, ', 100 - (data['gender_rate'] * 12.5), '% Male.\n'})
-		msg += ('Capture Rate: ', data['capture_rate'], " of 255. (Higher is better.)")
-		msg += ('Base Happiness: '), data['base_happiness'], '\n')
-		msg += ('Base Steps to Hatch: '), data['hatch_counter'] * 255 + 1, '\n')
-		msg += ("Growth Rate: "), data['growth_rate']['name'].title, '\n')
-		msg += ('Color and Shape: '), data['color']['name'].title, ' ', data['shape']['name'], '\n')
-		msg += ('Habitat: '), data['habitat']['name'].title, '\n')
-		msg += ('First seen in Generation ')
+		if data['gender_rate'] != -1:
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: Bad indent

**Comment by ValynEritai** [comment]
And fixed.

File: chat-bot/plugins/pokedex.py [hunk 13]
```diff
@@ -54,29 +53,29 @@ class AnimuAndMango(Plugin):
...
+            msg += ('Genderless'\n')
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: Syntax error

File: chat-bot/plugins/pokedex.py [hunk 14]
```diff
@@ -54,29 +53,29 @@ class AnimuAndMango(Plugin):
...
+		else:
+            msg += ('Gender Ratio:', data['gender_rate'] * 12.5, '% Female, ', 100 - data['gender_rate'] * 12.5, '% Male.\n')
+        msg += ('Capture Rate: ', data['capture_rate'], " of 255. (Higher is better.)")
```
Comment (cookkkie) [inline chat-bot/plugins/pokedex.py]: do not use (...) . To concatenate strings just use +

**Comment by ValynEritai** [comment]
Indents fixed, and I'll get right on the concatenation.

**Comment by ValynEritai** [comment]
And concatenation fixed. One more pass and we should be right to go.

**Comment by cookkkie** [comment]
You've got some remaining ) and (
+
a equal cond is not `=` but `==`
Gosh, I should make a CONTRIBUTE.md to show how to run mee6 locally for tests purpose :smile:

**Comment by ValynEritai** [comment]
Hahaha yeah. I'll fix one more time.

**Comment by ValynEritai** [comment]
Once again, hopefully fixed. There may be one error with the hatch counter line, but the +1 is meant to go into the hatch counter * 255, and I'm not sure if it will try and concatenate or not, so best to be safe. hehe

**Comment by ValynEritai** [comment]
Would you like me to add the files like PR #1 has to make it easier to import?

File: chat-bot/plugins/pokedex.py [hunk 15]
```diff
@@ -58,21 +58,21 @@ class Pokedex(Plugin):
         else:
              msg += 'Gender Ratio:' + data['gender_rate'] * 12.5 + '% Female + ' + 100 - data['gender_rate'] * 12.5 + '% Male.\n'
         msg += 'Capture Rate: ' + data['capture_rate'] + " of 255. (Higher is better.)"
-        msg += 'Base Happiness: ') + data['base_happiness'] + '\n'
-        msg += 'Base Steps to Hatch: ') + data['hatch_counter'] * 255 + 1 + '\n'
```
Comment (ValynEritai) [inline chat-bot/plugins/pokedex.py]: Quick question. Is thins line going to make the result hatch counter * 255 + 1 (Say, 10 * 255 + 1 = 2551) or will it make the result hatch counter * 255 with a 1 concatenated (Say, 10 * 255 = 2550 + 1 = 25501)?

File: chat-bot/plugins/pokedex.py [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+            return
+
+        msg = 'Pokemon Number' + data['id'] + ": " + data['names']['name'] + '\n'
+        if data['gender_rate'] != -1:
+            msg += 'Genderless\n'
+        else:
+             msg += 'Gender Ratio:' + data['gender_rate'] * 12.5 + '% Female + ' + 100 - data['gender_rate'] * 12.5 + '% Male.\n'
+        msg += 'Capture Rate: ' + data['capture_rate'] + " of 255. (Higher is better.)"
+        msg += 'Base Happiness: ' + data['base_happiness'] + '\n'
+        msg += 'Base Steps to Hatch: ' + data['hatch_counter'] * 255 + 1 + '\n'
```
Comment (ValynEritai) [inline chat-bot/plugins/pokedex.py]: Quick question. Is thins line going to make the result hatch counter * 255 + 1 (Say, 10 * 255 + 1 = 2551) or will it make the result hatch counter * 255 with a 1 concatenated? (Say, 10 * 255 = 2550 + 1 = 25501)

**Comment by KyleJHille** [comment]
So how do I use this? lol

**Comment by ValynEritai** [comment]
It needs to be added to the bot, which @cookkkie hasn't done yet.

**Comment by KyleJHille** [comment]
Figured. But the way the convo left off I assumed he maybe did it. :)

**Comment by cookiezeater** [comment]
You're pulling static data from pokedex server on every message. This is not recommended by pokedex devs. Add some caching :3 I think every resource has a limit of 200/day, so you may 403 pretty fast.

**Comment by ValynEritai** [comment]
I honestly cannot be bothered keeping this pull request open anymore. The code clearly needs work.

**Pull request closed without merging**